src/cluster.c

```diff
@@ -5224,7 +5224,7 @@ void migrateCommand(client *c) {
             socket_error = 1;
             break;
         }
-        if ((password && buf0[0] == '-') ||
+        if ((password && buf0[0] == '-' && !strstr(buf0, "but no password is set")) ||
             (select && buf1[0] == '-') ||
```

Comment:
A similar problem might happen if `select` fails because the target `dbid` doesn't exist in the target server.
I think the only decent solution for this is not to pipeline `auth`/`select` commands to the `restore` commands and verify their reply before continuing.